src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -353,12 +370,16 @@ private void EventQueueTaskAction()
                         // This should prevent a rare (but theoretically possible) scenario caused by context switching.
                         if (Interlocked.CompareExchange(ref _eventQueueTaskState, EventQueueTaskStopped, EventQueueTaskRunning) == EventQueueTaskRunning)
                             break;
+
+                    } else // if empty, let the thread sleep for a specified number of milliseconds before attempting to retrieve another value from the queue
+                    {
+                        Thread.Sleep(_eventQueuePollingInterval);
```

Comment:
Necessary to avoid continuous execution of the loop above when the queue is empty.